### ISSUE TITLE
2025-04-27 14:29:12,233 - ERROR - Error processing audio: Argument #4: Padding size should be less than the corresponding input dimension, but got: padding (2, 2) at dimension 2 of input [1, 80, 1]

### ISSUE THREAD

**Issue by WhozKunal** [m-bain/whisperX#1123]
I give 3 hour audio file when i process that for whisperx i got that error

2025-04-27 14:29:12,233 - ERROR - Error processing audio: Argument #4: Padding size should be less than the corresponding input dimension, but got: padding (2, 2) at dimension 2 of input [1, 80, 1]

 try:
            model = whisperx.load_model("medium", device, language="en", compute_type=compute_type)

            audio = whisperx.load_audio(temp_audio_file_path)

            result_transcribe = model.transcribe(audio, batch_size)

            model_a, metadata = whisperx.load_align_model(language_code=result_transcribe["language"], device=device)
        
            result_align = whisperx.align(result_transcribe["segments"],model_a,metadata,audio,device,return_char_alignments=False,)

            diarize_model = whisperx.DiarizationPipeline(use_auth_token=api_key, device=device)

            diarize_segments= diarize_model(audio, min_speakers, max_speakers)
            
            # Filter out segments shorter than 2 seconds
            
            result_final = whisperx.assign_word_speakers(diarize_segments, result_align)

            
    
            audio_segment_dir_sample = tempfile.mkdtemp()

            output_lines_speaker_wise = {"messages": []}

            segment_batches = defaultdict(list)
            sorted_segments = sorted(result_final["segments"], key=lambda x: x["start"])

            

            for segment_info in sorted_segments:
                speaker_label = segment_info.get("speaker", "Unknown")
                text = segment_info["text"]
                text = convert_special_words(text)
                start = segment_info["start"]
                end = segment_info["end"]

#Diazation
import numpy as np
import pandas as pd
from pyannote.audio import Pipeline
from typing import Optional, Union
import torch

from .audio import load_audio, SAMPLE_RATE

class DiarizationPipeline:
    def __init__(
        self,
        model_name="pyannote/speaker-diarization-3.1",
        use_auth_token=None,
        device: Optional[Union[str, torch.device]] = "CUDA",
    ):
        if isinstance(device, str):
            device = torch.device(device)
            
            
        self.model = Pipeline.from_pretrained(model_name, use_auth_token=use_auth_token).to(device)

    def __call__(self, audio: Union[str, np.ndarray], num_speakers=None, min_speakers=None, max_speakers=None):
        if isinstance(audio, str):
            audio = load_audio(audio)
            
            
            
            
            
        # write by kunal
        # audio_tensor = torch.from_numpy(audio[None, :])  # shape: [1, T]
        # min_len = int(SAMPLE_RATE * 2)  # 2 seconds
        # if audio_tensor.shape[-1] < min_len:
        #     print(f"⚠️ Skipping short audio segment: {audio_tensor.shape[-1] / SAMPLE_RATE:.2f} seconds")
        #     return pd.DataFrame()  # or return None if preferred

            
            
            
        audio_data = {'waveform': torch.from_numpy(audio[None, :]),'sample_rate': SAMPLE_RATE}
        
        
        
        # Perform diarization
        segments = self.model(audio_data, num_speakers = num_speakers, min_speakers=min_speakers, max_speakers=max_speakers)
        diarize_df = pd.DataFrame(segments.itertracks(yield_label=True), columns=['segment', 'label', 'speaker'])
        diarize_df['start'] = diarize_df['segment'].apply(lambda x: x.start)
        diarize_df['end'] = diarize_df['segment'].apply(lambda x: x.end)
        
        # write by Kunal
        # diarize_df = diarize_df[(diarize_df["end"] - diarize_df["start"]) >= 2.0]
        # print(f"Skipping short audio segment: {diarize_df['end'].sum() - diarize_df['start'].sum()} seconds")
        # write by Kunal

        return diarize_df

def assign_word_speakers(diarize_df, transcript_result, fill_nearest=False):
    transcript_segments = transcript_result["segments"]
    for seg in transcript_segments:
        # assign speaker to segment (if any)
        diarize_df['intersection'] = np.minimum(diarize_df['end'], seg['end']) - np.maximum(diarize_df['start'], seg['start'])
        diarize_df['union'] = np.maximum(diarize_df['end'], seg['end']) - np.minimum(diarize_df['start'], seg['start'])
        # remove no hit, otherwise we look for closest (even negative intersection...)
        if not fill_nearest:
            dia_tmp = diarize_df[diarize_df['intersection'] > 0]
        else:
            dia_tmp = diarize_df
        if len(dia_tmp) > 0:
            # sum over speakers
            speaker = dia_tmp.groupby("speaker")["intersection"].sum().sort_values(ascending=False).index[0]
            seg["speaker"] = speaker
        
        # assign speaker to words
        if 'words' in seg:
            for word in seg['words']:
                if 'start' in word:
                    diarize_df['intersection'] = np.minimum(diarize_df['end'], word['end']) - np.maximum(diarize_df['start'], word['start'])
                    diarize_df['union'] = np.maximum(diarize_df['end'], word['end']) - np.minimum(diarize_df['start'], word['start'])
                    # remove no hit
                    if not fill_nearest:
                        dia_tmp = diarize_df[diarize_df['intersection'] > 0]
                    else:
                        dia_tmp = diarize_df
                    if len(dia_tmp) > 0:
                        # sum over speakers
                        speaker = dia_tmp.groupby("speaker")["intersection"].sum().sort_values(ascending=False).index[0]
                        word["speaker"] = speaker
        
    return transcript_result            

class Segment:
    def __init__(self, start, end, speaker=None):
        self.start = start
        self.end = end
        self.speaker = speaker

**Comment by Barabazs** [comment]
Thanks for sharing this! It’s a bit hard to understand the issue since the problem statement is missing some details and the code block is quite large and unformatted. Could you please clarify what specific problem you’re seeing and share only relevant code (if needed)

Once updated, I’ll be happy to take another look!
